external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSerdeSuite.scala

```diff
@@ -86,18 +86,18 @@ class AvroSerdeSuite extends SparkFunSuite {
     // deserialize should have no issues when 'bar' is nullable but fail when it is nonnull
     Deserializer.create(CATALYST_STRUCT, avro, BY_NAME)
     assertFailedConversionMessage(avro, Deserializer, BY_NAME,
-      "Cannot find non-nullable field 'foo.bar' (at position 0) in Avro schema.",
+      "Cannot find field 'foo.bar' in Avro schema",
```

Comment:
Shall we mention "by-name" match here?
```
Cannot find field 'foo.bar' in Avro schema with by-name match
```

Reply:
Since by-name is the default, I'm thinking to omit that, and only mention if it's using positional matching. WDYT?